tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/contact/EmailAddressContact.scala

```diff
@@ -29,21 +34,40 @@ case class AccountEmailContact(accountId: String, id: UUID, address: MailAddress
 trait EmailAddressContactSearchEngine {
   def index(accountId: AccountId, address: MailAddress): Publisher[EmailAddressContact]
 
+  def index(accountId: AccountId, address: MailAddress, firstname: String, surname: String): Publisher[EmailAddressContact]
```

Comment:
This method has too much parameters, we likely should define  helper case class:

```
case class ContactFields(address: MailAddress, firstname: String, surname: String)
```

So that this then becomes:

```
def index(accountId: AccountId, fields: ContactFields): Publisher[EmailAddressContact]
```